core/controlplane/config/templates/cloud-config-controller

```diff
@@ -1,3 +1,21 @@
+{{ define "instance" -}}
+{{- $S3URI := self.Parts.s3.Asset.URL -}}
```

Comment:
Idea behind `self` was to give access part template to a whole UserData object, so that it doesn't need to depend on things like `.UserDataController.Parts` etc. Whole object is accessible so that if interface grows more complex, all those methods are available for template to use. Right now it is just `self.Parts` though, which I agree looks redundant.

`.Parts` is an artifact of previous iterations on this design. Now that it is settled, I might as well just drop it all together and go with `type UserData map[string]UserDataPart`, then it would be `self.s3.Asset.URL`,  what would you say?